src/entities/fractions/fraction.rs

```diff
@@ -108,23 +118,30 @@ where
     }
 
     // Helper method for converting any superclass back to a simple Fraction
-    fn as_fraction(&self) -> Fraction {
+    fn as_fraction(&self) -> Result<Fraction, Error> {
         Fraction::new(self.numerator().clone(), self.denominator().clone())
+            .map_err(|err| Error::CreateFractionalError(format!("{}", err)))
```

Comment:
```suggestion
    fn as_fraction(&self) -> Fraction {
        Fraction::new(self.numerator().clone(), self.denominator().clone()).unwrap()
    }
```
It's impossible to error when converting an existing fraction to fraction.